File: remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        result = []
        for i in range(len(s)):
            if result and s[i] == result[-1]:
                result.pop()
            else:
                result.append(s[i])
        return ''.join(result)

File: remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.js
//https://leetcode.com/problems/remove-all-adjacent-duplicates-in-string/

var removeDuplicates = function(s) {
    const result = [];
    result.push(s[0]);
    for (let i = 1; i < s.length; i++) {
        if (s[i] === result[result.length - 1]) {
            result.pop()
        } else {
            result.push(s[i])
        }
    };
    return result.join('')
};